File: 최경열/완전탐색/블랙잭/permutation.py
n, m = map(int, input().split(" "))
arr = list(map(int, input().split(" ")))
result = 0
for x in range(n):
    for y in range(x+1, n):
        for z in range(y+1, n):
            if arr[x] + arr[y] + arr[z] > m:
                continue
            else:
                result = max(result, arr[x] + arr[y] + arr[z])
print(result)

File: 최경열/완전탐색/블랙잭/permutation.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});
let input = [];
rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  const a = input[0].split(" ").map(Number);
  const n = a[0];
  const m = a[1];
  const arr = input[1].split(" ").map(Number);
  let result = 0;
  for (let x = 0; x < arr.length; x++) {
    for (let y = x + 1; y < arr.length; y++) {
      for (let z = y + 1; z < arr.length; z++) {
        if (arr[x] + arr[y] + arr[z] > m) {
          continue;
        } else {
          result = Math.max(result, arr[x] + arr[y] + arr[z]);
        }
      }
    }
  }
  console.log(result);
});
